invenio/modules/oauthclient/contrib/orcid.py

```diff
@@ -122,12 +123,38 @@ def account_info(remote, resp):
 
 def account_setup(remote, token):
     """Perform additional setup after user have been logged in."""
+    from invenio.modules.accounts.models import UserEXT
     from invenio.ext.sqlalchemy import db
     from ..handlers import token_session_key
 
+    import json
+    import requests
+
+    orcid = session.get(token_session_key(remote.name) +
+                        "_account_info").get("external_id")
+
     extra_data = {
-        "orcid": session.get(token_session_key(remote.name) +
-                             "_account_info").get("external_id")
+        "orcid": orcid
         }
     token.remote_account.extra_data = extra_data
     db.session.commit()
+
+    # Fill user full name if not already set
+    user = UserEXT.query.filter_by(id=orcid).first().user
```

Comment:
For my education, what is the difference between the joined query and accessing the user with the backref as it is now?